src/melee.cpp

```diff
@@ -322,11 +292,10 @@ void player::melee_attack(Creature &t, bool allow_special, const matec_id &force
 
         t.on_dodge( this, get_melee() );
 
-        if( !has_active_bionic("bio_cqb") ) {
-            // No practice if you're relying on bio_cqb to fight for you
-            melee_practice( *this, false, unarmed_attack(),
-                            weapon.is_bashing_weapon(), weapon.is_cutting_weapon(),
-                            (weapon.has_flag("SPEAR") || weapon.has_flag("STAB")));
+        // Practice melee and relevant weapon skill (if any) except when using CQB bionic
+        if( !has_active_bionic( "bio_cqb" ) ) {
+            practice( skill_melee, rng( 5, 10 ) );
+            practice( weapon.weap_skill(), rng( 5, 10 ) );
```

Comment:
Can you look at what I wrote (d1438d8) which defacto results in either 1 or 2 weapon skills being trained plus melee.